### ISSUE TITLE
Async Await Test setup

### ISSUE THREAD

**Issue by snewcomer** [wyeworks/ember-qunit-nice-errors#18]
Seems to not work if tests setup w/ async await.  Any thoughts?

**Comment by fedekau** [comment]
@snewcomer Can you please provide an example of a test where the addon doesn't do it's magic? That's always a helpful and nice thing to have :grinning:

**Comment by snewcomer** [comment]
@fedekau Here is the 
Here is the acceptance test.  Seems to be somewhere linked to the `TestTransformFilter`.

https://github.com/snewcomer/ember-qunit-nice-errors/commits/master

![screen shot 2016-07-21 at 4 31 45 pm](https://cloud.githubusercontent.com/assets/7374640/17042243/d12c71ee-4f60-11e6-893f-ed6fd9c16578.png)

**Comment by fedekau** [comment]
@snewcomer Thanks, we will take a look at this!

**Comment by les2** [comment]
I can reproduce in my project. Tried updating and a unit testing that uses `assert.async()` fails.

Beginning of test has this:

```
  let done = assert.async();
  assert.expect(7);

```

Failure message is this:

```
    ✘ Expected assert.expect(7) assertions, but 7 were run
            at testWrapper (http://localhost:7357/assets/addon-test-support/ember-qunit/test-wrapper.js:44:1)
            at test (http://localhost:7357/assets/addon-test-support/ember-qunit/test.js:11:1)
            at test (http://localhost:7357/assets/performance/tests/ember-sinon-qunit/test.js:31:1)
```

I gotta run to a kids birthday part soon but later I'll try to distill to a reproducible example.

**Comment by les2** [comment]
@fedekau minimal ember project demonstrating the bug: https://github.com/les2/ember-qunit-nice-errors-issue-18

(see readme) or just `git log` to see what happens from `ember init`.

let me know if there or questions.

now going to that party! getting yelled at now!

**Comment by fedekau** [comment]
@les2 I am sorry to hear that. The are two issues that can aid with this problem, see #20 & #35. They will not fix the error, but at least they will unblock you from using the addon 😄 

The problem, we believe, is related with the parser we are using, which is [recast](https://github.com/benjamn/recast) which does not support  `aync` and `await`.

We are planing to use a different one, for example [babylon](https://github.com/babel/babylon).

I will take a look at the example, thanks for that!

Have fun, and don't eat too much 🍰 !

**Comment by sdebarros** [comment]
@fedekau @les2 I think the issue in this case is different than the one reported first, it's not caused by calling `assert.async`, but by enabling `completeExistingMessages`. Created a new issue for this https://github.com/wyeworks/ember-qunit-nice-errors/issues/38

**Comment by san650** [comment]
@fedekau @acostami @sdebarros I'm rewriting the transformation as a Babel Plugin, I have a working version already but I have to clean up the code first before submitting the PR. The advantage of having this as a babel plugin is that it will support any notation supported by Babylon (the babel parser).

So if you add the babel async functions to your project, this addon will work because we will use the same parser.
